File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/enums/YGEdge.java
package io.github.orioncraftmc.meditate.internal.enums;

public enum YGEdge {
    YGEdgeLeft,
    YGEdgeTop,
    YGEdgeRight,
    YGEdgeBottom,
    YGEdgeStart,
    YGEdgeEnd,
    YGEdgeHorizontal,
    YGEdgeVertical,
    YGEdgeAll;

    public static final int SIZE = java.lang.Integer.SIZE;

    public static YGEdge forValue(int value) {
        return values()[value];
    }

    public int getValue() {
        return this.ordinal();
    }
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/internal/enums/YGEdge.kt
package app.cash.redwood.yoga.internal.enums

enum class YGEdge {
  YGEdgeLeft,
  YGEdgeTop,
  YGEdgeRight,
  YGEdgeBottom,
  YGEdgeStart,
  YGEdgeEnd,
  YGEdgeHorizontal,
  YGEdgeVertical,
  YGEdgeAll;

  fun getValue(): Int {
    return ordinal
  }

  companion object {
    fun forValue(value: Int): YGEdge {
      return values()[value]
    }
  }
}
